src/core/server/opensearch/version_check/ensure_opensearch_version.test.ts

```diff
@@ -244,7 +247,11 @@ describe('pollOpenSearchNodesVersion', () => {
       });
   });
 
-  it('starts polling immediately and then every opensearchVersionCheckInterval', () => {
+  /*
+   * Disable TestScheduler test due to OOM issue,
+   * refer to https://rxjs.dev/guide/testing/marble-testing#known-issues
+   */
+  it.skip('starts polling immediately and then every opensearchVersionCheckInterval', () => {
```

Comment:
I also forgot to add - I feel like we should have new unit tests for this feature.

Reply:
I skipped only because I found this issue and wasn't sure what next steps would be but didn't want to make too many changes in this PR since it doesn't directly relate to the healthcheck code.

Will do on unit tests.

Reply:
If we can't address it in this PR and there isn't a known solution can we at least add a `TODO`?

Reply:
Sure. Will add a TODO